### ISSUE TITLE
How to detect when the app is removed from recent apps in Flutter?

### ISSUE THREAD

**Issue by GowthamanRavichandran3** [flutter/flutter#153908]
### Steps to reproduce

1. Run the provided sample,
2. Then close the app from recent apps,

### Expected results

While closing the app from recent apps, **AppLifecycleState.detached** should be triggered.

### Actual results

While closing the app from recent apps, **AppLifecycleState.paused** was triggered. But **AppLifecycleState.paused** was also triggered when switching the app.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> with WidgetsBindingObserver {
  @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addObserver(this);
  }

  @override
  void dispose() {
    WidgetsBinding.instance.removeObserver(this);
    super.dispose();
  }

  @override
  void didChangeAppLifecycleState(AppLifecycleState state) {
    if (state == AppLifecycleState.detached) {
      // App was closed from recent tabs
    }
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('My App'),
        ),
        body: Center(
          child: Text('Hello, World!'),
        ),
      ),
    );
  }
}

```

However, AppLifecycleState.detached is never triggered when I remove the app from the recent apps list. How can I detect this scenario correctly? Is there an alternative approach or a solution to this problem?

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.19.4, on Microsoft Windows [Version 10.0.19045.4780], locale en-US)
    • Flutter version 3.19.4 on channel stable at D:\Flutter SDK\flutter     
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 68bfaea224 (5 months ago), 2024-03-20 15:36:31 -0700
    • Engine revision a5c24f538d
    • Dart version 3.3.2
    • DevTools version 2.31.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\GowthamanRavichandra\AppData\Local\Android\sdk     
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio1\jbr\bin\java      
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[!] Android Studio (version 2021.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter       
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X Unable to determine bundled Java version.
    • Try updating or re-installing Android Studio.

[√] Android Studio (version 2022.3)
    • Android Studio at C:\Program Files\Android\Android Studio1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)

[√] VS Code, 64-bit edition (version 1.85.0)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.94.0

[√] Connected device (4 available)
    • moto g32 (mobile) • ZD222FBQYC • android-arm64  • Android 13 (API 33)
    • Windows (desktop) • windows    • windows-x64    • Microsoft Windows [Version 10.0.19045.4780]
    • Chrome (web)      • chrome     • web-javascript • Google Chrome 127.0.6533.120
    • Edge (web)        • edge       • web-javascript • Microsoft Edge 127.0.2651.105

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @GowthamanRavichandran3, what device are you seeing this on? I just tested on my Pixel 7 and android emulator and it worked as expected on both devices. 

You also appear to be using an older version of flutter, can you upgrade to the latest version of flutter stable to see if you still experience this issue? If you do, please provide the updated output of `flutter doctor -v`. 

Thank you

**Comment by GowthamanRavichandran3** [comment]
``> Hi @GowthamanRavichandran3, what device are you seeing this on? I just tested on my Pixel 7 and android emulator and it worked as expected on both devices.
> 
> You also appear to be using an older version of flutter, can you upgrade to the latest version of flutter stable to see if you still experience this issue? If you do, please provide the updated output of `flutter doctor -v`.
> 
> Thank you

Hi @danagbemava-nc, Thanks for your response. 

Yes, I have used the older version of Flutter. But even after updating to the most recent version, the issue persists.. Please find the **flutter doctor - v** output below.

`[√] Flutter (Channel stable, 3.24.1, on Microsoft Windows [Version 10.0.19045.4780], locale en-US)
    • Flutter version 3.24.1 on channel stable at D:\Flutter SDK\flutter   
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5874a72aa4 (2 days ago), 2024-08-20 16:46:00 -0500
    • Engine revision c9b9d5780d
    • Dart version 3.5.1
    • DevTools version 2.37.2

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\GowthamanRavichandra\AppData\Local\Android\sdk     
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio1\jbr\bin\java      
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[!] Android Studio (version 2021.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X Unable to determine bundled Java version.
    • Try updating or re-installing Android Studio.

[√] Android Studio (version 2022.3)
    • Android Studio at C:\Program Files\Android\Android Studio1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)

[√] VS Code, 64-bit edition (version 1.85.0)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.94.0

[√] Connected device (4 available)
    • moto g32 (mobile) • ZD222FBQYC • android-arm64  • Android 13 (API 33)
    • Windows (desktop) • windows    • windows-x64    • Microsoft Windows [Version 10.0.19045.4780]
    • Chrome (web)      • chrome     • web-javascript • Google Chrome 127.0.6533.120
    • Edge (web)        • edge       • web-javascript • Microsoft Edge 127.0.2651.105

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.`

Thanks.

**Comment by danagbemava-nc** [comment]
Hi @GowthamanRavichandran3, are you only seeing this on the Moto g32? What other devices have you tested on?

**Comment by GowthamanRavichandran3** [comment]
> Hi @GowthamanRavichandran3, are you only seeing this on the Moto g32? What other devices have you tested on?

Hi @danagbemava-nc, I have tested in Moto g32, Pixel C (Android emulator) and Pixel 7 (Android emulator). Not working in the mentioned devices.

**Comment by danagbemava-nc** [comment]
Hi @GowthamanRavichandran3, do you get this issue when you move to a different app and remove the app from the list or do you even get this issue when you move the app to recents and remove it from the list?

**Comment by GowthamanRavichandran3** [comment]
> AppLifecycleState.paused

Hi @danagbemava-nc, **AppLifecycleState.paused** was triggered when the app was switched on and when it was closed, but **AppLifecycleState.detached** was never triggered.

**Comment by danagbemava-nc** [comment]
What are the exact steps you take to reproduce the issue? Please bear in mind there isn't a guarantee that flutter will receive the signal from android. See https://github.com/flutter/flutter/issues/57594#issuecomment-1231965622

**Comment by mohammadamini-tooti** [comment]
Hi
I'm using flutter lifecycles in my app to check when the user is online or when they become offline and for that I'm listening to onPaused, onResume and onInActive states. Everything works fine on lower versions of android and when the user opens recent activity, both on InActive and OnPaused is called so I can use onPaused() to listen to whether the user is active or not and if the user attempts to close the app through recent activity onPaused() is called. But in later versions of Android like 14, 15 or maybe even below when the user opens the recent activity screen to kill the app, onPaused is not called so I have to use onInactive to send a signal to firebase that the user is trying to kill the app cause there is no trigger that checks if the user actually swipes and kills the app. But the problem with this approach is that everytime a navigation is done in the app the onInactive is called for a brief moment and then onResume is called right after that so that user's status changes to offline for a brief moment.

In short, there is no lifecycle event that checks if the user terminates the app and onInActive is called whenever the app state changes even as much as a navigation.
